tests/extendrtests/src/rust/build.rs

```diff
@@ -1,12 +1,56 @@
-use std::env;
+use std::path::PathBuf;
 
 fn main() {
-    // TODO: I couldn't find any nice way to add the condition based on the R version
-    // except for using libR-sys just for "DEP_R_*" envvars.
-    let major = env::var("DEP_R_R_VERSION_MAJOR").unwrap();
-    let minor = env::var("DEP_R_R_VERSION_MINOR").unwrap();
+    println!("cargo:rerun-if-env-changed=R_HOME");
+    println!("cargo::rerun-if-changed=build.rs");
 
-    if &*major >= "4" && &*minor >= "3" {
+    let r_home_env = std::env::var_os("R_HOME");
+    let r_home_env = r_home_env.map(PathBuf::from);
+
+    // prefer `R_HOME` if set, otherwise use the `PATH` available R
+    let mut r_version = if let Some(r_home_env) = r_home_env {
```

Comment:
Why do you call this `r_version`? While it seems like it is a path to R executable

Reply:
Renamed.